### PR TITLE
fix issue #271 config ConnectTimeout and ReadTimeout separately

### PR DESCRIPTION
**Pull request by Hugo-X**
# Checklist
- [x] I have read the [contribution guidelines] (https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CONTRIBUTING.md).
- [x] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- If this is a modification that impacts the behavior of a public API
  - [ ] I edited the corresponding document in the `devdoc` folder and added or modified requirements.
- I submitted this PR against the correct branch: 
  - [x] This pull-request is submitted against the `master` branch. 

# Reference/Link to the issue solved with this PR (if any)
https://github.com/Azure/azure-iot-sdk-java/issues/271

# Description of the problem

https://github.com/Azure/azure-iot-sdk-java/issues/271

# Description of the solution
 
add setConnectTimeoutMillis(int timeout) to com.microsoft.azure.sdk.iot.service.transport.http.HttpRequest

### CODE REVIEW

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceOperations.java [hunk 1]
```diff
@@ -80,10 +108,10 @@ public static HttpResponse request(
             throw new IllegalArgumentException("Null payload");
         }
 
-        /* Codes_SRS_DEVICE_OPERATIONS_99_018: [The request shall throw IllegalArgumentException if the provided `timeoutInMs` exceed Integer.MAX_VALUE.] */
-        if((timeoutInMs + DEFAULT_HTTP_TIMEOUT_MS) > Integer.MAX_VALUE) 
+        /* Codes_SRS_DEVICE_OPERATIONS_99_018: [The request shall throw IllegalArgumentException if the provided `responseTimeoutInMs` exceed Integer.MAX_VALUE.] */
+        if((responseTimeoutInMs) > Integer.MAX_VALUE)
         {
-            throw new IllegalArgumentException("HTTP Request timeout shouldn't not exceed " + timeoutInMs + DEFAULT_HTTP_TIMEOUT_MS + " milliseconds");
+            throw new IllegalArgumentException("HTTP Request timeout shouldn't not exceed " + Integer.MAX_VALUE +  " milliseconds");
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/DeviceOperations.java]: This same check is needed for connectTimeoutInMs as well as it is long to int conversion

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java [hunk 2]
```diff
@@ -42,7 +42,9 @@
     private IotHubConnectionString iotHubConnectionString;
     private URL url;
     private HttpMethod httpMethod;
-    private long timeout;
+
+    private long connectTimeoutInMs;
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java]: As per our coding guidelines, please remove extra lines line 45

File: service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java [hunk 3]
```diff
@@ -142,23 +144,42 @@ private void continueQuery(String continuationToken, int pageSize) throws IOExce
...
+     * @param responseTimeoutInMs Maximum time to wait for the hub to respond
      * @return QueryResponse object which holds the response Iterator
      * @throws IOException If any of the input parameters are not valid
      * @throws IotHubException If HTTP response other then status ok is received
      */
     public QueryResponse sendQueryRequest(IotHubConnectionString iotHubConnectionString,
                                    URL url,
                                    HttpMethod method,
-                                   Long timeoutInMs) throws IOException, IotHubException
+                                   Long connectTimeoutInMs,
```
Comment (prmathur-microsoft) [inline service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java]: This is a breaking change as you are modifying a public API. My suggestion is to leave the existing API as is and create a new one with additional parameters

Comment (Hugo-X) [reply service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/devicetwin/Query.java]: keep the existing API, but change the parameter name _timeoutInMs_ to _responseTimeoutInMs_

**Comment by jasmineymlo** [comment]
Merge was attempted but this PR broke some e2e tests and is currently blocked at the gate.  Investigation is needed.  Thanks.

**Comment by Hugo-X** [comment]
> Merge was attempted but this PR broke some e2e tests and is currently blocked at the gate. Investigation is needed. Thanks.

Can I get the details of those tests?  
Thank you.

**Pull request closed without merging**